### DISCUSSION TITLE
SCIP callbacks and multithreading

### DISCUSSION THREAD

**Discussion by CADBIMDeveloper** [google/or-tools#2612] [Linear Solver questions]
Hi, @Mizux 

You've recently [enabled multithreading in SCIP solver](https://github.com/google/or-tools/commit/846e49c04ee04f70129d05d33cb8a301edcf7dbc), thank you for that. Could you also fix it if solver callback is set? Or at least take a look, please. Perhaps it is not a google or-tools, but SCIP solver issue.

Thank you!

**Comment by lperron** [comment]
What is the problem with callbacks ?

**Reply by CADBIMDeveloper** [reply]
Hi,

When I started this discussion I had 2 issues, the first is obivous - failed assertion in concsolver_scip on line 260 after SCIPcopy invocation.

But it seems it works in Release configuration... Don't know what to expect...

My second problem was that my application worked instable when I started multiple solvers with callbacks in parallel with my [exposed to .Net callbacks](https://github.com/google/or-tools/pull/2608). However, I fixed this today. The problem was related to GC. My callback object was collected and callback invocation failed
